asv_bench/benchmarks/period.py

```diff
@@ -96,6 +96,8 @@ def time_value_counts(self, typ):
 class Indexing(object):
 
     goal_time = 0.2
+    pi_with_nas = PeriodIndex(['1985Q1', 'NaT', '1985Q2'] * 1000, freq='Q')
+    pi_diff = PeriodIndex(['1985Q1', 'NaT', '1985Q3'] * 1000, freq='Q')
```

Comment:
`setup` gets called with every round of `time_foo`.  Doing this here is analogous to calling `setup_class` once.

Reply:
I _think_ this is what `setup_cache` is for: https://asv.readthedocs.io/en/stable/writing_benchmarks.html#setup-and-teardown-functions

Reply:
setup_cache pickles and then loads in each step of the loop